docs/resources/server/advancements.md

```diff
@@ -0,0 +1,173 @@
+
+Advancement Criteria
+--------------------
+
+To unlock an advancement, the specified criteria must be met. Criteria are tracked through triggers which execute when a certain action is performed: killing an entity, changing an inventory, breading animals, etc. Any time an advancement is loaded into the game, the criteria defined are read and added as listeners to the trigger. Afterwards a trigger function is called (usually named `#trigger`) which checks all listeners as to whether the current state meets the conditions of the advancement criteria. The criteria listeners for the advancement are only removed once the advancement has been obtained by completing all requirements.
+
+Requirements are defined as an array of string arrays representing the name of the criteria specified on the advancement. An advancement is completed once one string array of criteria has been met:
+
+```js
+// In some advancement JSON
+
+// List of defined criteria to meet
+"criteria": {
+  "example_criterion1": { /*...*/ },
+  "example_criterion2": { /*...*/ },
+  "example_criterion3": { /*...*/ },
+  "example_criterion4": { /*...*/ }
+},
+
+// This advancement is only unlocked once
+// - Criteria 1 AND 2 have been met
+// OR
+// - Criteria 3 and 4 have been met
+"requirements": [
+  [
+    "example_criterion1",
+    "example_criterion2"
+  ],
+  [
+    "example_criterion3",
+    "example_criterion4"
+  ]
+]
+```
+
+A list of criteria triggers defined by vanilla can be found in `CriteriaTriggers`. Additionally, the JSON formats are defined on the [Minecraft Wiki][triggers].
+
+### Custom Criteria Triggers
+
+Custom criteria triggers can be created by implementing `SimpleCriterionTrigger` for the created `AbstractCriterionTriggerInstance` subclass.
+
+### AbstractCriterionTriggerInstance Subclass
+
+The `AbstractCriterionTriggerInstance` represents a single criteria defined in the `criteria` object. Trigger instances are responsible for holding the defined conditions, returning whether the inputs match the condition, and writing the instance to JSON for data generation.
+
+Conditions are usually passed in through the constructor. The `AbstractCriterionTriggerInstance` super constructor requires the instance to define the registry name of the trigger and the conditions the player must meet as an `EntityPredicate$Composite`. The registry name of the trigger should be supplied to the super directly while the conditions of the player should be a constructor parameter.
+
+```java
+// Where ID is the registry name of the trigger
+public ExampleTriggerInstance(EntityPredicate.Composite player, ItemPredicate item) {
+  super(ID, player);
+  // Store the item condition that must be met
+}
+```
+
+!!! note
+    Typically, trigger instances have a static constructor which allow these instances to be easily created for data generation.
```

Comment:
It should likely be noted that another (likely) reason for the existence of these static factory methods is to allow static import of these methods in the relevant data providers.